components/suggestion/modal_suggestion_list.jsx

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import SuggestionList from 'components/suggestion/suggestion_list.jsx';
+import {getClosestParent} from 'utils/utils.jsx';
+
+export default class ModalSuggestionList extends React.PureComponent {
+    static propTypes = {
+        location: PropTypes.string.isRequired,
+        calculateInputRect: PropTypes.func.isRequired,
+        onLoseVisibility: PropTypes.func.isRequired,
+    }
+
+    latestHeight = 0;
+    scrollRegistered = false;
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            scroll: 0,
+            modalBounds: {top: 0, bottom: 0},
+            inputBounds: {top: 0, bottom: 0, width: 0},
+            location: props.location,
+        };
+
+        this.container = React.createRef();
+    }
+
+    tryRegisterScroll = () => {
+        if (this.container.current) {
+            const modalBodyContainer = getClosestParent(ReactDOM.findDOMNode(this.container.current), '.modal-body');
+            modalBodyContainer.addEventListener('scroll', this.onModalScroll);
+            this.scrollRegistered = true;
+        }
+    }
+
+    onModalScroll = (e) => {
+        if (this.state.scroll !== e.target.scrollTop &&
+            this.latestHeight !== 0) {
+            this.setState({scroll: e.target.scrollTop});
+        }
+    }
+
+    componentDidMount() {
+        this.tryRegisterScroll();
+        window.addEventListener('resize', this.updateModalBounds);
+    }
+
+    componentWillUnmount() {
+        if (this.container.current && this.scrollRegistered) {
+            const modalBodyContainer = getClosestParent(ReactDOM.findDOMNode(this.container.current), '.modal-body');
+            modalBodyContainer.removeEventListener('scroll', this.onModalScroll);
+        }
+        window.removeEventListener('resize', this.updateModalBounds);
+    }
+
+    componentDidUpdate() {
+        if (!this.scrollRegistered) {
+            this.tryRegisterScroll();
+        }
+
+        if (this.getChildHeight() === 0) {
+            return;
+        }
+
+        const newInputBounds = this.updateInputBounds();
```

Comment:
Generally, we want code in `componentDidUpdate` to only trigger when there's some sort of change in the props or state since since updates can happen due to unrelated state changes. Can this be wrapped in something like `if (this.state.scroll !== prevState.scroll) {...}`?

Reply:
Using only the scroll would not be enough. Resizing the screen (and therefore changing the modal bounds) may not change the scroll, and that would not work. Also, a change in the child size (that means the selector is opened) may not change the scroll, which will also not work.

Nevertheless, I like the idea of only doing these changes whenever they are needed. I will upload an approach on the next commit.